src/DebugChannel.js

```diff
@@ -26,22 +26,26 @@ export class DebugChannel {
   /*
   /* Only called in the beginning to check if a debugger is attached
   */
-  checkForDebugger(): boolean {
+  debuggerIsAttached(): boolean {
     let line = this.readIn();
     if (line === "Debugger Attached\n") {
       this.writeOut("Ready\n");
       return true;
     }
     return false;
   }
+
   /* Reads in a request from the debug adapter
   /* The caller is responsible for sending a response with the appropriate
   /* contents at the right time.
   /* For now, it returns the request as a string. It will be made to return a
   /* Request object based on the protocol
   */
   readIn(): string {
-    let contents = this.fs.readFileSync(this.inFilePath, "utf8").toString();
+    let contents = "";
+    while (contents.length === 0) {
+      contents = this.fs.readFileSync(this.inFilePath, "utf8").toString();
```

Comment:
why the call to toString?

Reply:
I originally thought readFileSync would return a Buffer but it looks like specifying the encoding will already turn the result into a string so I will remove this.